### ISSUE TITLE
design: there's no timeout for out/lock workflow

### ISSUE THREAD

**Issue by pro-wh** [oasisprotocol/oasis-bridge#72]
if the witnesses go offline or for whatever other reason refuse to process a lock event, there's no escape hatch to cancel the transfer and recover the locked amount

the same is probably true for the other direction, as in/release workflow only handles forward progress

**Comment by kostko** [comment]
Yes, I'm not sure how this could be handled absent having a dynamic witness set / governance which we tried to avoid in v1.

**Comment by pro-wh** [comment]
right... and there's no cross-chain way to prove what order a pair of cancel/witness events happened in
